source/extensions/filters/http/cache/cache_filter.cc

```diff
@@ -30,11 +31,10 @@ struct CacheResponseCodeDetailValues {
 
 using CacheResponseCodeDetails = ConstSingleton<CacheResponseCodeDetailValues>;
 
-CacheFilter::CacheFilter(const envoy::extensions::filters::http::cache::v3::CacheConfig& config,
-                         const std::string&, Stats::Scope&, TimeSource& time_source,
-                         HttpCache& http_cache)
-    : time_source_(time_source), cache_(http_cache),
-      vary_allow_list_(config.allowed_vary_headers()) {}
+CacheFilter::CacheFilter(const std::string&, Stats::Scope&, TimeSource& time_source,
+                         const CacheFilterConfig& config)
+    : config_(config), cache_(nullptr), time_source_(time_source),
```

Comment:
I think it's more envoy-style to do like `=nullptr` in the header file for things to be initialized that don't require parameters.